metricflow/model/semantics/semantic_model_lookup.py

```diff
@@ -41,27 +41,31 @@
 
 
 class SemanticModelLookup(SemanticModelAccessor):
```

Comment:
Unrelated to this PR, but I'm wondering if we need these ABC classes (`SemanticModelAccessor` and `MetricAccessor`). Seems like we could get rid of them and just use private functions to restrict what callers use. Though nothing is truly private in python. Or maybe protocols would be better.

Reply:
Fun fact, these are Protocols, not ABCs, we just inherit from them because it makes it easier to find the relationship. This was from before the ProtocoloHint thing.

That aside, I had the exact same thought - the Protocols were built to ensure callers would never add models or whatever but that is no longer a meaningful concern. So I was actually thinking about doing this, but then I ran into the other bit about the protocol, which was how it allowed us to sidestep some circular import dependency problems, so I gave up for now.

But yeah, whenever we feel so inclined we should probably just ditch the interface classes.

Reply:
Heck yeah 💥